src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.EnumerateSplits.cs

```diff
@@ -0,0 +1,270 @@
+            EnumerateSplits(input, count, startat: RightToLeft ? input.Length : 0);
+
+        /// <summary>
+        /// Searches an input span for all occurrences of a regular expression and returns a <see cref="ValueSplitEnumerator"/> to iterate over the splits around matches.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// The behavior of <see cref="EnumerateSplits(ReadOnlySpan{char}, int, int)"/> is similar to the behavior of <see cref="Split(string, int, int)"/>, producing the splits
+        /// one at a time as part of iterating through the resulting enumerator rather than all at once as part of a single array. However, there are a few notable differences.
+        /// <see cref="Split(string, int, int)"/> will include the contents of capture groups in the resulting splits, while <see cref="EnumerateSplits(ReadOnlySpan{char}, int, int)"/> will not.
+        /// And if <see cref="RegexOptions.RightToLeft"/> is specified, <see cref="Split(string, int, int)"/> will reverse the order of the resulting splits to be left-to-right, whereas
+        /// <see cref="EnumerateSplits(ReadOnlySpan{char}, int, int)"/> will yield the splits in the order they're found right-to-left.
+        /// </para>
+        /// <para>
+        /// Each match won't actually happen until <see cref="ValueSplitEnumerator.MoveNext"/> is invoked on the enumerator, with one match being performed per <see cref="ValueSplitEnumerator.MoveNext"/> call.
+        /// Since the evaluation of the match happens lazily, any changes to the passed in input in between calls to <see cref="ValueSplitEnumerator.MoveNext"/> will affect the match results.
+        /// </para>
+        /// </remarks>
+        /// <param name="input">The span to search for a match.</param>
+        /// <param name="count">The maximum number of times the split can occur. If 0, all splits are available.</param>
+        /// <param name="startat">The zero-based character position at which to start the search.</param>
+        /// <returns>A <see cref="ValueSplitEnumerator"/> to iterate over the matches.</returns>
+        public ValueSplitEnumerator EnumerateSplits(ReadOnlySpan<char> input, int count, int startat)
+        {
+            if (count < 0)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.count, ExceptionResource.CountTooSmall);
+            }
+
+            if ((uint)startat > (uint)input.Length)
+            {
+                ThrowHelper.ThrowArgumentOutOfRangeException(ExceptionArgument.startat, ExceptionResource.BeginIndexNotNegative);
+            }
+
+            return new ValueSplitEnumerator(this, input, count, startat, RightToLeft);
+        }
+
+        /// <summary>
+        /// Represents an enumerator containing the set of splits around successful matches found by iteratively applying a regular expression pattern to the input span.
+        /// </summary>
+        [StructLayout(LayoutKind.Auto)]
```

Comment:
any particular reason to put this -- is it our pattern to put this on every struct that has no particular reason (ie interop) to be laid out sequentially?

(do we have/should we have an analyzer to ensure we have StructLayout on all structs, to be explicit?)

Reply:
> do we have/should we have an analyzer to ensure we have StructLayout on all structs, to be explicit?

https://github.com/dotnet/runtime/issues/64261

> any particular reason to put this

Just to avoid me having to arrange the fields optimally, and just seeing from the list of fields that they're not.